### PR TITLE
Update revdeps script to do hashing and cleaning up our requirements files.

### PR DESCRIPTION
**Pull request by EnTeQuAk**

### CODE REVIEW

File: scripts/hash_requirements.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
+import pip
+import sys
+from pip.req import parse_requirements
+from hashin import run as run_hashin
```
Comment (diox) [inline scripts/hash_requirements.py]: You're missing the change that adds hashin to our requirements

Comment (diox) [inline scripts/hash_requirements.py]: Also, not in alphabetical order :)

File: scripts/hash_requirements.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    with open(requirements_path, 'a') as fobj:
+        fobj.write('\n')
+
+    for req in requirements:
+        run_hashin(
+            '%s%s' % (req.name, req.specifier),
+            requirements_path,
+            'sha256',
+            python_versions=['py27', '2.7'],
```
Comment (diox) [inline scripts/hash_requirements.py]: Add comment mentioning that it works around an issue with hashin

**Comment by diox** [comment]
r+wc

**Comment by diox** [comment]
Now I remember why I did not do the rewriting of the requirements.txt file in that script: it does not work so well with requirements including other requirements with `-r`...

**Comment by EnTeQuAk** [comment]
This is the diff that is now produced running the script over `dev.txt`

```diff
diff --git a/requirements/dev.txt b/requirements/dev.txt
index 6f78d89..51c7d0d 100644
--- a/requirements/dev.txt
+++ b/requirements/dev.txt
@@ -1,32 +1,46 @@
---no-binary Pillow
 -r prod.txt
 -r docs.txt
 -r flake8.txt
 
-## Testing related dependencies
+# coverage is required by pytest-cov
+coverage==4.0.3 \
+    --hash=sha256:94c1e66610807a7917d967ed6415b9d5fde7487ab2a07bb5e054567865ef6ef0 \
+    --hash=sha256:76a73a48a308fb87a4417d630b0345d36166f489ef17ea5aa8e4596fb50a2296 \
+    --hash=sha256:beb96d32ce8cfa47ec6433d95a33e4afaa97c19ac1b4a47ea40a424fedfee7c2 \
+    --hash=sha256:85b1275b6d7a61ccc8024a4e9a4c9e896394776edce1a5d075ec116f91925462 \
+    --hash=sha256:addf63b5e39d573c459c3930b25176146395c1dc1afce4710067bb5e6dc4ea58 \
+    --hash=sha256:d3188345f1c7161d701fd2ea9150f9bb6e2df890f3ddd6c0aea1f525e21d1544 \
+    --hash=sha256:00d464797a236f654337181af72b4baea3d35d056ca480e45e9163bb5df496b8
+django-debug-toolbar==1.3.2 \
+    --hash=sha256:6c1353b68a158642e14860f7278b81755e87bbda89e353673862c9079475d103 \
+    --hash=sha256:d05765591dce6a00aa5fc66179c5cfcfd9540f92046ce8a9e37cf3d4b8d5fe17
+django-fixture-magic==0.0.4 \
+    --hash=sha256:33bffe85f63cb55aa1f31696003a9e5055aa280019651455a4fcdec0842a9acc
 # execnet is required by pytest-cache, pytest-xdist
-execnet==1.2 --hash=sha256:16293a69b17c846371f1ccb9cef10ad88838c9d5239ba26d88b39efb7b432f6f \
-             --hash=sha256:951714caa0ae80237f4ffc1f08450e9e2e5f8f902beaf1ad294020875d6f8c2c
-psutil==0.2.0 --hash=sha256:a103e3f9d07b7fabcf00d1f1c1c28607ba37e41a7e8e63a61bf62acfa02f506c \
-              --hash=sha256:8de8efa92162c94f623297f522e440e818dc7b832f421f9f490324bc7d5d0d92
-# py is required by pytest, pytest-xdist
-py==1.4.30 --hash=sha256:b703e57685ed7c280b1a51c496a4984d83d89def2a930b5e9e5da5a6ca151514 \
-           --hash=sha256:07e20ab90a550bd3c21891e0d887f0931b4098f148aec95e29b5188f161bb075
-# pytest is required by pytest-django, pytest-instafail, pytest-cache, pytest-xdist
-pytest==2.7.0 --hash=sha256:91dc842785417208d57e5e8dc8bb40f57316c45da24a50c53b49cb8d045519d6
-pytest-cache==1.0 --hash=sha256:495ec1b5e9112f5bfb7c6bd75ac5e5b56b41e025fdb5afe21b191be31d3fb961 \
-                  --hash=sha256:be7468edd4d3d83f1e844959fd6e3fd28e77a481440a7118d430130ea31b07a9
-pytest-django==2.8.0 --hash=sha256:d145ac9dc7a557a719ab79770be0941004e1e038e137c34591919d9df2a790b1 \
-                     --hash=sha256:d76f934e77fa073f48cc521945a49900a859e610fa029dd880d1d8b997b77c23
-pytest-xdist==1.13.1 --hash=sha256:4382d7a944c1e2b1dc17fee3fd3575495fb10236a3b6e95cc6955db04984bc3c \
-                     --hash=sha256:e76f59b69fc0ccde8b9477284a7fb726a24a21f5faae6ab3059392e8bbd88aa7
-pytest-cov==2.2.1 --hash=sha256:40bfd01002ceaf9e6c5d371d3c4d541d378a4214ea03e45f934980766a809384
-coverage==4.0.3 --hash=sha256:85b1275b6d7a61ccc8024a4e9a4c9e896394776edce1a5d075ec116f91925462
-
-# General development helpers
-django-debug-toolbar==1.3.2 --hash=sha256:d05765591dce6a00aa5fc66179c5cfcfd9540f92046ce8a9e37cf3d4b8d5fe17
-django-fixture-magic==0.0.4 --hash=sha256:8ca514d1f52e57475ffc744e93161eb50848dcec245a16c06d8c5b7b5497c305 \
-                            --hash=sha256:33bffe85f63cb55aa1f31696003a9e5055aa280019651455a4fcdec0842a9acc
+execnet==1.2.0 \
+    --hash=sha256:951714caa0ae80237f4ffc1f08450e9e2e5f8f902beaf1ad294020875d6f8c2c
+psutil==0.2.0 \
+    --hash=sha256:8de8efa92162c94f623297f522e440e818dc7b832f421f9f490324bc7d5d0d92 \
+    --hash=sha256:4ce48cbbe3c915a9b720ce6465da34ce79465ffde63de8254cc9c8235fef824d \
+    --hash=sha256:08c6f16ac00babe2c54067dbc305c0587cb896a8eb64383290cdafdba74ea123
+# py is required by pytest, pytest-xdist, tox
+py==1.4.30 \
+    --hash=sha256:07e20ab90a550bd3c21891e0d887f0931b4098f148aec95e29b5188f161bb075 \
+    --hash=sha256:b703e57685ed7c280b1a51c496a4984d83d89def2a930b5e9e5da5a6ca151514
+# pytest is required by pytest-cache, pytest-cov, pytest-django, pytest-xdist
+pytest==2.7.0 \
+    --hash=sha256:91dc842785417208d57e5e8dc8bb40f57316c45da24a50c53b49cb8d045519d6
+pytest-cache==1.0 \
+    --hash=sha256:be7468edd4d3d83f1e844959fd6e3fd28e77a481440a7118d430130ea31b07a9
+pytest-cov==2.2.1 \
+    --hash=sha256:a8b22e53e7f3b971454c35df99dffe21f4749f539491e935c55d3ff7e1b284fa \
+    --hash=sha256:40bfd01002ceaf9e6c5d371d3c4d541d378a4214ea03e45f934980766a809384
+pytest-django==2.8.0 \
+    --hash=sha256:d145ac9dc7a557a719ab79770be0941004e1e038e137c34591919d9df2a790b1 \
+    --hash=sha256:d76f934e77fa073f48cc521945a49900a859e610fa029dd880d1d8b997b77c23
+pytest-xdist==1.13.1 \
+    --hash=sha256:4382d7a944c1e2b1dc17fee3fd3575495fb10236a3b6e95cc6955db04984bc3c \
+    --hash=sha256:e76f59b69fc0ccde8b9477284a7fb726a24a21f5faae6ab3059392e8bbd88aa7
 # sqlparse is required by django-debug-toolbar
-sqlparse==0.1.14 --hash=sha256:36f0f40af9ef4d358b963ab275f62331083c68290a3d14a2199d5dd4d98cc45d \
-                 --hash=sha256:e561e31853ab9f3634a1a2bd53035f9e47dfb203d56b33cc6569047ba087daf0
+sqlparse==0.1.14 \
+    --hash=sha256:e561e31853ab9f3634a1a2bd53035f9e47dfb203d56b33cc6569047ba087daf0
```

**Comment by diox** [comment]
Excellent. r+

**Pull request merged**